Pass conn ID to ObjectStoragePath via URI

uranusjr opened the pull request:
This enables an alternative ObjectStoragePath init syntax, using the auth section in the URI to supply conn ID instead of a separate keyword argument. The explicit keyword argument is honored if supplied.

uranusjr committed:
Ensure fs attached in tests are cleaned up

uranusjr committed:
Pass conn ID to ObjectStoragePath via URI

This enables an alternative ObjectStoragePath init syntax, using the
auth section in the URI to supply conn ID instead of a separate keyword
argument. The explicit keyword argument is honored if supplied.

uranusjr committed:
Add explaination on connection ID in tutorial

uranusjr committed:
Add note on parsing logic

hussein-awala reviewed (approved):
It looks good! I hope we will not have more complicated needs in the future that conflict with passing the connection ID in the user part.

uranusjr committed:
Do not use two different parser

While unlikely, it is *possible* for CPython to change how username and
password is parsed from netloc. To ensure we always do the right thing,
it is better to just implement the logic outselves so it always work as
expected. We need this logic to get the host info anyway, so it is not
possible to solely rely on CPython's parser.
